### PR TITLE
Upgrade from Chromium 101 to Chromium 102.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes https://github.com/brave/brave-browser/issues/21976

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-browser-profiles-profile_manager.h.patch [hunk 1]
```diff
@@ -8,5 +8,5 @@ index cc3c0dfb5be1b31a7dfbc252350a19d62f39d1b8..abf2b024fedcb7c8929a711d47d88d1b
    }
 +  BRAVE_PROFILE_MANAGER_H
```
Comment (goodov) [inline patches/chrome-browser-profiles-profile_manager.h.patch:9]: I don't think this is a correct place for this line now, it should go right before `private:`.

**Review by simonhong — APPROVED** [review]
browser/ui/views/infobars ++

File: chromium_src/base/win/default_apps_util.cc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#define BRAVE_IPFS L"ipfs"
+#define BRAVE_IPNS L"ipns"
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/base/win/default_apps_util.cc:7]: Done.

File: patches/chrome-browser-profiles-profile_manager.h.patch (hunk 1 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-profiles-profile_manager.h.patch]: Fixed in 4855be50917ac30097f470baef638e37531ddcf4

File: chromium_src/chrome/browser/chrome_content_browser_client.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_CHROME_CONTENT_BROWSER_CLIENT_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_CHROME_CONTENT_BROWSER_CLIENT_H_
+
+#define CHROME_CONTENT_BROWSER_CLIENT_H \
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/chrome_content_browser_client.h]: Fixed in 8a846d7b390707528dd2c49dd4b64854ceb1cdbf

File: patches/chrome-browser-browsing_data-chrome_browsing_data_remover_delegate.cc.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc b/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc
+index de599532eca93890d09b8da60b03c365239a526a..cd22e707599e6fc8787d9bbd064440b5bc6d9109 100644
+--- a/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc
++++ b/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc
+@@ -1359,6 +1359,7 @@ const char* ChromeBrowsingDataRemoverDelegate::GetHistogramSuffix(
+       return "WebAppHistory";
+     case TracingDataType::kWebAuthnCredentials:
+       return "WebAuthnCredentials";
++    BRAVE_BROWSER_TRACING_DATA_TYPE_IPFS_CACHE
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply patches/chrome-browser-browsing_data-chrome_browsing_data_remover_delegate.cc.patch]: Fixed in ef0f38751ecf39ae0ae54b7f09ce1d464a952f5e

File: patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch [hunk 5]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/ui/tabs/tab_strip_model.cc b/chrome/browser/ui/tabs/tab_strip_model.cc
+index 415d3605ff3be8a80396ec7c7ea4ac3f3b71e9cb..800adac46aa100abc60fe903d1280b674b5c0209 100644
+--- a/chrome/browser/ui/tabs/tab_strip_model.cc
++++ b/chrome/browser/ui/tabs/tab_strip_model.cc
+@@ -1240,6 +1240,7 @@ void TabStripModel::RemoveFromGroup(const std::vector<int>& indices) {
+ 
+ bool TabStripModel::IsReadLaterSupportedForAny(
+     const std::vector<int>& indices) {
++  BRAVE_TAB_STRIP_MODEL_DEACTIVATE_READING_LIST
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch]: Fixed in 567bc90816a7db0e85ef34a8ae1b2f1bba389f43

File: patches/chrome-browser-ui-views-frame-browser_view.cc.patch [hunk 6]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/ui/views/frame/browser_view.cc b/chrome/browser/ui/views/frame/browser_view.cc
+index c20c5ed30291fc59b91c72c64bf17d276bcaa137..e750eb9fccfbf9e7fafcf9b60818cff170350d2c 100644
+--- a/chrome/browser/ui/views/frame/browser_view.cc
++++ b/chrome/browser/ui/views/frame/browser_view.cc
+@@ -2179,6 +2179,7 @@ void BrowserView::OnFeatureEngagementTrackerInitialized(bool initialized) {
+   if (!initialized)
+     return;
+   MaybeShowWebUITabStripIPH();
++  BRAVE_BROWSER_VIEW_DEACTIVATE_READING_LIST
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-views-frame-browser_view.cc.patch]: Fixed in 5f24ea5aee9e94f4e582c0e0d1fed373c584fbd1

File: chromium_src/chrome/browser/ui/tabs/tab_strip_model.cc [hunk 7]
```diff
@@ -0,0 +1,13 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/reading_list/features/reading_list_switches.h"
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/tabs/tab_strip_model.cc]: Fixed in 1e91acc6e0919ffb636c11e82ee022be78b1ba21

File: chromium_src/chrome/browser/ui/views/frame/browser_view.cc [hunk 8]
```diff
@@ -7,11 +7,16 @@
 #include "brave/browser/ui/views/tabs/brave_tab_strip.h"
 #include "brave/browser/ui/views/toolbar/brave_toolbar_view.h"
 #include "chrome/browser/ui/views/frame/tab_strip_region_view.h"
+#include "components/reading_list/features/reading_list_switches.h"
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/views/frame/browser_view.cc]: Fixed in 567bc90816a7db0e85ef34a8ae1b2f1bba389f43

File: chromium_src/chrome/browser/ui/views/toolbar/toolbar_view.cc [hunk 9]
```diff
@@ -18,8 +18,14 @@
       AddChildView(std::make_unique<BraveAvatarToolbarButton>(browser_view_)); \
   if (false)
 
+#define BRAVE_TOOLBAR_DEACTIVATE_SIDE_PANEL \
+  if (true)                                 \
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/views/toolbar/toolbar_view.cc]: Fixed in f3e5f4e9b295b658a18ef946bca7396bdac69aec

File: chromium_src/base/win/default_apps_util.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/base/win/default_apps_util.cc]: Fixed in 353111f6087852ee45760e2bad6ab10ee227a6a0

File: patches/chrome-browser-ui-views-toolbar-toolbar_view.cc.patch [hunk 10]
```diff
@@ -1,8 +1,16 @@
...
-index 8dceab5d386ec7bd3cc51c8ba2d29136854948fa..40f6daadf453959e8cc5ae03babc8eddcc7824ca 100644
+index 8dceab5d386ec7bd3cc51c8ba2d29136854948fa..ea8c7d38d68939dd5962a659a609358fd0dd8a4d 100644
 --- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
 +++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
-@@ -370,6 +370,7 @@ void ToolbarView::Init() {
+@@ -302,6 +302,7 @@ void ToolbarView::Init() {
+   }
+ 
+   std::unique_ptr<SidePanelToolbarButton> side_panel_button;
++  BRAVE_TOOLBAR_DEACTIVATE_SIDE_PANEL
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-views-toolbar-toolbar_view.cc.patch]: Fixed in e119794f2fb5b8909b0066c2e9bbc7a8a62d7cab

File: chromium_src/chrome/browser/ui/tabs/tab_strip_model.h [hunk 11]
```diff
@@ -8,8 +8,12 @@
 
 #define SelectRelativeTab virtual SelectRelativeTab
 #define TAB_STRIP_MODEL_H_ friend class BraveTabStripModel;
+#define IsReadLaterSupportedForAny     \
+  IsReadLaterSupportedForAny_Unused(); \
```
Comment (goodov) [inline chromium_src/chrome/browser/ui/tabs/tab_strip_model.h:12]: a simpler replacement like `virtual IsReadLaterSupportedForAny` doesn't work here?

File: chromium_src/content/browser/browser_interface_binders.cc [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+#undef PopulateFrameBinders
+
+namespace content {
+namespace internal {
+
+void PopulateFrameBinders(RenderFrameHostImpl* host, mojo::BinderMap* map) {
+  PopulateFrameBinders_ChromiumImpl(host, map);
+
+  // Keep handwriting recognition disabled
+  map->Add<handwriting::mojom::HandwritingRecognitionService>(
```
[the remark this answers is not in the document]
Comment (goodov) [reply chromium_src/content/browser/browser_interface_binders.cc:17]: @emerick @cdesouza-chromium have you tried to call the JS that requests this mojo interface? is everything fine there?

**Review by samartnik — APPROVED** [review]
++

File: chromium_src/content/browser/browser_interface_binders.cc (hunk 12 above)
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/content/browser/browser_interface_binders.cc:17]: I'm gonna have an answer in a min.

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: chromium_src/chrome/browser/ui/tabs/tab_strip_model.h (hunk 11 above)
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/tabs/tab_strip_model.h]: Fixed in 5ecd21a2fdf2b38cd2ec1eb377fa7c0b2cbea7e1

**Review by kylehickinson — APPROVED** [review]
iOS++

File: chromium_src/content/browser/browser_interface_binders.cc (hunk 12 above)
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/content/browser/browser_interface_binders.cc:17]: I have tried to make use of these API (e.g. navigator.createHandwritingRecognizer) to no avail. It seems to me they are deactivated.

[the remark this answers is not in the document]
Comment (bridiver) [reply chromium_src/content/browser/browser_interface_binders.cc:17]: @cdesouza-chromium the normal way we'd disable this is by overriding handwriting_recognition_service.cc in chromium_src. See BatteryManager

File: ios/browser/providers/BUILD.gn [hunk 13]
```diff
@@ -17,6 +17,7 @@ group("brave_providers") {
     "//ios/chrome/browser/providers/branded_images:chromium_branded_images",
     "//ios/chrome/browser/providers/discover_feed:chromium_discover_feed",
     "//ios/chrome/browser/providers/lens:chromium_lens",
+    "//ios/chrome/browser/providers/mailto_handler:chromium_mailto_handler",
```
Comment (bridiver) [inline ios/browser/providers/BUILD.gn:20]: cc @kylehickinson - want to make sure this doesn't affect any handlers in the main brave ios app

File: patches/components-os_crypt-os_crypt_win.cc.patch [hunk 14]
```diff
@@ -1,17 +1,17 @@
...
+index 91f2c6d4f79cf233770abbbcaaedbdd246ee044e..84b8a28a8f18a074ce02bdbdc9821693f22d854d 100644
 --- a/components/os_crypt/os_crypt_win.cc
 +++ b/components/os_crypt/os_crypt_win.cc
-@@ -146,6 +146,7 @@ bool OSCryptImpl::DecryptString16(const std::string& ciphertext,
- 
+@@ -177,6 +177,7 @@ bool OSCryptImpl::DecryptString16(const std::string& ciphertext,
+ // static
  bool OSCryptImpl::EncryptString(const std::string& plaintext,
                              std::string* ciphertext) {
 +  if (IsEncryptionDisabled(plaintext, ciphertext)) { return true; }
```
Comment (bridiver) [inline patches/components-os_crypt-os_crypt_win.cc.patch:9]: in the future please fix old style patches if you have to resolve conflicts on them

File: ios/browser/providers/BUILD.gn (hunk 13 above)
Comment (kylehickinson) [reply ios/browser/providers/BUILD.gn:20]: We should be fine, this provider only handles the actual mailto schemes when handled but it needs to be setup and called from the iOS side which we dont do

File: patches/chrome-browser-chrome_content_browser_client.h.patch [hunk 15]
```diff
@@ -1,12 +1,12 @@
...
-index f0415a5099cdf181ab620fb6400db4f524c2c892..75d763c33a425f21d0aa62e77fd37c27408338d7 100644
+index f0415a5099cdf181ab620fb6400db4f524c2c892..2200962fde512acac8b67f173a1af79599e18373 100644
 --- a/chrome/browser/chrome_content_browser_client.h
 +++ b/chrome/browser/chrome_content_browser_client.h
-@@ -804,6 +804,7 @@ class ChromeContentBrowserClient : public content::ContentBrowserClient {
+@@ -803,6 +803,7 @@ class ChromeContentBrowserClient : public content::ContentBrowserClient {
+     extra_parts_.push_back(part);
    }
  
++  CHROME_CONTENT_BROWSER_CLIENT_H
```
Comment (bridiver) [inline patches/chrome-browser-chrome_content_browser_client.h.patch]: `define InProcessBrowserTest InProcessBrowserTest; friend class BraveContentBrowserClient` - but what do we need this friend class for?

File: patches/chrome-browser-extensions-extension_service.h.patch [hunk 16]
```diff
@@ -1,12 +1,12 @@
...
-index 1265719e1b83f08e79827e871e13e15883aea508..aeb9809eea27f8d249c8737a179ca29b73b9afdd 100644
+index 1265719e1b83f08e79827e871e13e15883aea508..c6325944935e4f7d11b3fff0342e0e3bda472df6 100644
 --- a/chrome/browser/extensions/extension_service.h
 +++ b/chrome/browser/extensions/extension_service.h
-@@ -491,6 +491,7 @@ class ExtensionService : public ExtensionServiceInterface,
+@@ -490,6 +490,7 @@ class ExtensionService : public ExtensionServiceInterface,
+   // While disabled all calls to CheckForExternalUpdates() will bail out.
    static base::AutoReset<bool> DisableExternalUpdatesForTesting();
  
++  BRAVE_EXTENSION_SERVICE_H
```
Comment (bridiver) [inline patches/chrome-browser-extensions-extension_service.h.patch:9]: do not automatically replace these with another patch. There is often a `define` we can use like the other ones I flagged

File: patches/chrome-browser-extensions-extension_management.h.patch [hunk 17]
```diff
@@ -1,12 +1,12 @@
...
-index 1c4a68369bb5030e763a52b35573e7319d7fa641..13ccd5fd3175b4b551bea7c8170293b2a4d57430 100644
+index 1c4a68369bb5030e763a52b35573e7319d7fa641..b0c19b664a2a3c685e8e7c263a4d7d730a879d2d 100644
 --- a/chrome/browser/extensions/extension_management.h
 +++ b/chrome/browser/extensions/extension_management.h
-@@ -230,6 +230,7 @@ class ExtensionManagement : public KeyedService {
+@@ -229,6 +229,7 @@ class ExtensionManagement : public KeyedService {
+   // Returns whether the profile associated with this instance is supervised.
    bool is_child() const { return is_child_; }
  
++  BRAVE_EXTENSION_MANAGEMENT_H
```
Comment (bridiver) [inline patches/chrome-browser-extensions-extension_management.h.patch:9]: see below

File: patches/chrome-browser-profiles-profile_manager.h.patch [hunk 18]
```diff
@@ -1,12 +1,12 @@
...
-+  BRAVE_PROFILE_MANAGER_H
+@@ -355,6 +355,7 @@ class ProfileManager : public Profile::Delegate {
+       std::unique_ptr<AccountProfileMapper> mapper);
+ #endif
  
- #if BUILDFLAG(IS_CHROMEOS_LACROS)
-   void SetAccountProfileMapperForTests(
++  BRAVE_PROFILE_MANAGER_H
+  private:
+   friend class TestingProfileManager;
```
Comment (bridiver) [inline patches/chrome-browser-profiles-profile_manager.h.patch:11]: `#define TestingProfileManager TestingProfileManager; friend class BraveProfileManager;`

File: patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch (hunk 5 above)
Comment (bridiver) [inline patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch]: it looks like all of these patches were eventually removed? Is that correct?

File: components/content_settings/renderer/brave_content_settings_agent_impl.cc [hunk 19]
```diff
@@ -270,18 +271,22 @@ bool BraveContentSettingsAgentImpl::IsCosmeticFilteringEnabled(
 
 bool BraveContentSettingsAgentImpl::IsFirstPartyCosmeticFilteringEnabled(
     const GURL& url) {
+  if (!IsCosmeticFilteringEnabled(url))
```
Comment (iefremov) [inline components/content_settings/renderer/brave_content_settings_agent_impl.cc:274]: absense of this was a bug?

File: chromium_src/content/browser/browser_interface_binders.cc (hunk 12 above)
[the remark this answers is not in the document]
Comment (iefremov) [reply chromium_src/content/browser/browser_interface_binders.cc:17]: but @cdesouza-chromium can you please check brave://crashes/ aswell? it could generate `base::DumpWithoutCrashing()`

**Review by iefremov — APPROVED** [review]

**Review by petemill — APPROVED** [review]
I'm just approving the package.json change as a @brave/js-deps-reviewers member, but please let me know if there's anything else I should look at in detail

File: DEPS [hunk 20]
```diff
@@ -19,7 +19,7 @@ deps = {
   "third_party/bitcoin-core/src": "https://github.com/bitcoin/bitcoin.git@95ea54ba089610019a74c1176a2c7c0dba144b1c",
   "third_party/argon2/src": "https://github.com/P-H-C/phc-winner-argon2.git@62358ba2123abd17fccf2a108a301d4b52c01a7c",
   "third_party/boost/config": "https://github.com/boostorg/config.git@e108255ffb5d2557ed3398b3fc575a2e9fd434cc",
-  "third_party/boost/multiprecision": "https://github.com/boostorg/multiprecision.git@db0bb4d2d2611605b7ce97220fb9973fd7e072c9",
+  "third_party/boost/multiprecision": "https://github.com/boostorg/multiprecision.git@32aefd37f055cc2abeced1cd2873b4564ea339f2",
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply DEPS:22]: There was a build breakage due to dead code on boost.multiprecision. I submitted a simple patch to them, which has been merged since.

https://github.com/boostorg/multiprecision/pull/446

This code was not failing to build on before 102, which suggests the warning flag got activated upstream at some point.

File: patches/components-os_crypt-os_crypt_win.cc.patch (hunk 14 above)
Comment (cdesouza-chromium) [reply patches/components-os_crypt-os_crypt_win.cc.patch:9]: Noted. Thanks.

File: components/content_settings/renderer/brave_content_settings_agent_impl.cc (hunk 19 above)
Comment (cdesouza-chromium) [reply components/content_settings/renderer/brave_content_settings_agent_impl.cc]: They were added as part of a patch you had provided me. I have tested, and I can confirm that check in particular is not necessary and no tests are failing without it. I'm removing it in e5499c4da52e075b3e19d0077bed8377f9aab08d

File: patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch (hunk 5 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-tabs-tab_strip_model.cc.patch]: Yes, that's correct. I have squashed it all now under 4c129602728052de745fac1ff6121124c60d88f8

File: patches/chrome-browser-profiles-profile_manager.h.patch (hunk 18 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-profiles-profile_manager.h.patch]: Fixed in bc012a12237f8ce3735467e076624e77b00cca3f

File: patches/chrome-browser-extensions-extension_management.h.patch (hunk 17 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-extensions-extension_management.h.patch]: Fixed in b9fd8f07d86dfbccdfca4edc692bcd4044de2b33

File: patches/chrome-browser-chrome_content_browser_client.h.patch (hunk 15 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-chrome_content_browser_client.h.patch]: I had touched on these files trying to improve a pre-existing patch that existed in there, but now that you've brought up the question of whether or not the patch is required in the first place, I realise I had not thought of that. This patch might have been necessary at some point, but I can confirm it isn't necessary anymore.

Fixed in 54d73c820ce37e83b609fdc769dd936289de9615

File: patches/chrome-browser-browsing_data-chrome_browsing_data_remover_delegate.cc.patch (hunk 4 above)
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply patches/chrome-browser-browsing_data-chrome_browsing_data_remover_delegate.cc.patch]: Fixed in 46c8b357d8d4e48c22e135349c621d77282affc1

**Review by bridiver — APPROVED** [review]

**Review by goodov — APPROVED** [review]

**Comment by emerick** [comment]
CI passed on all platforms; ready to merge.

**Comment by mkarolin** [comment]
When uplifting, also uplift:
* https://github.com/brave/brave-browser/issues/22881
* https://github.com/brave/brave-browser/issues/22897
